### PR TITLE
[meta] [rpc] Use location in local inc headers for gensairpc.pl

### PR DESCRIPTION
**Pull request by kcudnik**
it may happen that metadata is already generated, and SAI direcoty is mounted in docker, then the location will be invalid inside docker, so let's use only filename and inc directory

### CODE REVIEW

File: meta/gensairpc.pl [hunk 1]
```diff
@@ -302,35 +302,46 @@ sub generate_server_template_from_skeleton {
 }
 
 # Find the api name within the header related to the currenct XML file.
-sub get_api_name {
+sub get_api_name
+{
+    # it may happen that metadata is already generated, and SAI direcoty is
```
Comment (chrispsommers) [inline meta/gensairpc.pl:307]: s/direcoty/directory/

File: meta/gensairpc.pl [hunk 2]
```diff
@@ -302,35 +302,46 @@ sub generate_server_template_from_skeleton {
...
-        close $header or croak;
     }
 
-    $api //= $1 if $location =~ /inc\/sai(\w+)[.]h$/;
-    $api //= 'common';
-    $api = 'common' if $api eq 'types';
-
-    $api_names{$location} = $api;
-    return $api;
+    die "File $file don't contain api struct!";
```
Comment (chrispsommers) [inline meta/gensairpc.pl:344]: s/don't/doesn't/

Comment (kcudnik) [reply meta/gensairpc.pl:344]: fixed https://github.com/opencomputeproject/SAI/pull/1870

File: meta/gensairpc.pl (hunk 1 above)
Comment (kcudnik) [reply meta/gensairpc.pl:307]: fixed https://github.com/opencomputeproject/SAI/pull/1870

**Pull request merged**